s2s/time.py

```diff
@@ -414,41 +412,108 @@ def __str__(self):
         return f"{self._n_intervals} periods of {self.freq} leading up to {self.month}/{self.day}."
 
     def __repr__(self):
-        props = ", ".join([f"{k}={v}" for k, v in self.__dict__.items() if not k.startswith('_')])
+        if self._intervals is not None:
+            return repr(self._intervals)
+
+        props = ", ".join(
+            [f"{k}={v}" for k, v in self.__dict__.items() if not k.startswith("_")]
+        )
+        return f"AdventCalendar({props})"
+
+    def _repr_html_(self):
+        """For jupyter notebook to load html compatiable version of __repr__."""
+        if self._intervals is not None:
+            return self._intervals._repr_html_() # pylint: disable=protected-access
+
+        props = ", ".join(
+            [f"{k}={v}" for k, v in self.__dict__.items() if not k.startswith("_")]
+        )
         return f"AdventCalendar({props})"
 
-    def discard(self, max_lag):
+    @property
+    def flat(self):
+        if self._intervals is not None:
+            return self._intervals.stack()
+        raise ValueError("The calendar is not initialized with intervals yet.")
+
+    def discard(self, max_lag):  # or "set_max_lag"
         """Only keep indices up to the given max lag."""
         # or think of a nicer way to discard unneeded info
         raise NotImplementedError
 
-    def mark_target_period(self, start=None, end=None, periods=None):
+    def mark_target_period(self, start=None, periods=None):
         """Mark indices that fall within the target period."""
         # eg in pd.period_range you have to specify 2 of 3 (start/end/periods)
-        if start and end:
-            pass
-        elif start and periods:
+        if start is not None:
             pass
-        elif end and periods:
+        elif periods:
             pass
         else:
-            raise ValueError("Of start/end/periods, specify exactly 2")
+            raise ValueError("Specify either start or periods, not both.")
+
         raise NotImplementedError
 
     def get_lagged_indices(self, lag=1):  # noqa
         """Return indices shifted backward by given lag."""
         raise NotImplementedError
 
-    def get_train_indices(self, strategy, params):  # noqa
-        """Return indices for training data indices using given strategy."""
-        raise NotImplementedError
+    @property
+    def traintest(self) -> pd.DataFrame:
+        """Shorthand for getting both train and test indices.
 
-    def get_test_indices(self, strategy, params):  # noqa
-        """Return indices for test data indices using given strategy."""
-        raise NotImplementedError
+        The user will get a flat intervals list similar to the output of `map_years`
+        or `map_data`, but with extra columns containing `train` and `test` labels.
```

Comment:
Document that the user must first set the method (using calendar.set_traintest_method)

Reply:
Add "The user must first set a method using `set_traintest_method`." to the docstrings.